impl(pubsub): add default pull lease manager class

alevenberg opened the pull request:
Part of https://github.com/googleapis/google-cloud-cpp/issues/13298

These are exact copies of the PullLeaseManager except with s/pull_lease_manager/default_pull_lease_manager and other variations.

In the next PR I will turn PullLeaseManager into an interface so the PullAckHandler can hold either a DefaultPullLeaseManager or a TracingPullLeaseManager{DefaultPullLeaseManager}

<!-- Reviewable:start -->
- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/googleapis/google-cloud-cpp/13355)
<!-- Reviewable:end -->

coryan commented on google/cloud/pubsub/internal/default_pull_lease_manager.cc:
```diff
@@ -0,0 +1,134 @@
+// Copyright 2022 Google LLC
```
nit: copyright year

alevenberg replied:
Ack, changed in the new PR

coryan reviewed (approved):
It may be clearer (when looking at the history of these files) if you *renamed* the files and classes in a single PR instead of copying them and then renaming things.

alevenberg commented:
I didn't know how to save this PR, but created a new one. TIL: `git mv`.

Moved the files then cherry picked the commit from this PR. Also, updated the copyright year.

https://github.com/googleapis/google-cloud-cpp/pull/13365
